js/src/views/Account/Transactions/store.js

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { action, observable, transaction } from 'mobx';
+
+import etherscan from '~/3rdparty/etherscan';
+
+export default class Store {
+  @observable address = null;
+  @observable isLoading = false;
+  @observable isTest = undefined;
+  @observable isTracing = false;
+  @observable txHashes = [];
+
+  constructor (api) {
+    this._api = api;
+  }
+
+  @action setHashes = (transactions) => {
+    transaction(() => {
+      this.setLoading(false);
+      this.txHashes = transactions.map((transaction) => transaction.hash);
+    });
+  }
+
+  @action setAddress = (address) => {
+    this.address = address;
+  }
+
+  @action setLoading = (isLoading) => {
+    this.isLoading = isLoading;
+  }
+
+  @action setTest = (isTest) => {
+    this.isTest = isTest;
+  }
+
+  @action setTracing = (isTracing) => {
+    this.isTracing = isTracing;
+  }
+
+  @action updateProps = (props) => {
+    transaction(() => {
+      this.setAddress(props.address);
+      this.setTest(props.isTest);
+
+      // TODO: When tracing is enabled again, adjust to actually set
+      this.setTracing(false && props.traceMode);
+    });
+
+    return this.getTransactions();
+  }
+
+  getTransactions () {
+    if (this.isTest === undefined) {
+      return Promise.resolve();
+    }
+
+    this.setLoading(true);
+
+    // TODO: When supporting other chains (eg. ETC). call to be made to other endpoints
```

Comment:
Agreed. Didn't want to do it here since it may be one of those exploding tasks - however would like to start using netChain instead of isTest as soon as possible.